incubator/hnc/pkg/controllers/config_controller_test.go

```diff
@@ -52,6 +54,29 @@ var _ = Describe("HNCConfiguration", func() {
 		Eventually(hasSecret(ctx, barName, "foo-sec")).Should(BeTrue())
 		Expect(secretInheritedFrom(ctx, barName, "foo-sec")).Should(Equal(fooName))
 	})
+
+	It("should propagate an empty object whose types have been added to HNCConfiguration", func() {
+		setParent(ctx, barName, fooName)
+		// Creates an empty ConfigMap. We use ConfigMap for the test because the apiserver will not
+		// add additional fields to an empty ConfigMap object to make it non-empty.
+		makeConfigMap(ctx, fooName, "foo-config")
+
+		// Wait 1 second to give "foo-config" a chance to be propagated to bar, if it can be propagated.
+		time.Sleep(1 * time.Second)
+		// Foo should have "foo-config" since we created it there.
+		Eventually(hasConfigMap(ctx, fooName, "foo-config")).Should(BeTrue())
+		// "foo-config" is not propagated to bar because ConfigMap hasn't been configured in HNCConfiguration.
+		Eventually(hasConfigMap(ctx, barName, "foo-config")).Should(BeFalse())
+
+		Eventually(func() error {
+			c := getHNCConfig(ctx)
+			return addConfigMapToHNCConfig(ctx, c)
+		}).Should(Succeed())
+
+		// "foo-config" should now be propagated from foo to bar.
+		Eventually(hasConfigMap(ctx, barName, "foo-config")).Should(BeTrue())
+		Expect(configMapInheritedFrom(ctx, barName, "foo-config")).Should(Equal(fooName))
```

Comment:
This is the only "real" part of this test - everything else is tested elsewhere. Maybe create a helper to add ConfigMaps as a supported type in the object controller (e.g. change `addConfigMapToHNCConfig` to include the whole `Eventually` and `getHNCCConfig` stuff) so you can call it from object-controller-test, and skip the stuff where you try to propagate _before_ creating the object controller, since you already test that.

Reply:
Thanks for the suggestion! PTAL